test/client.py

```diff
@@ -0,0 +1,555 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import mock
+import uuid
+import redis
+import socket
+import mockredis
+import unittest
+import logging
+from mockredis import mock_strict_redis_client
+
+from phonon.client import Client
+from phonon.exceptions import ClientError
+from phonon.client.config import configure
+from phonon.client.config.node import Node
+from phonon.operation import Operation
+from phonon.exceptions import ReadError, EmptyResult, WriteError, Rollback
+
+from phonon import logger
+
+console = logger.get_logger(__name__, log_level=logging.INFO)
+
+
+def mock_redis(f):
+
+    @mock.patch('redis.StrictRedis', mock_strict_redis_client)
+    def wrapped(*args, **kwargs):
+        return f(*args, **kwargs)
+
+    wrapped.__name__ = f.__name__
+
+    return wrapped
+
+
+class ClientTest(unittest.TestCase):
+
+    def setUp(self):
+        configure({'ec': ['A', 'B'],
+                   'wc': ['C', 'D']})
+
+        self.client = Client()
+
```

Comment:
Given nodes A, B, and C; write a test to ensure we roll back (_forward_) to the latest committed among the three (A) in the event of a failure on the other two (B, C) preventing commit.